api/v1alpha2/conversion.go

```diff
@@ -79,13 +80,26 @@ func (dst *ClusterList) ConvertFrom(srcRaw conversion.Hub) error {
 func (src *Machine) ConvertTo(dstRaw conversion.Hub) error {
 	dst := dstRaw.(*v1alpha3.Machine)
 
+	// Manually restore data.
+	restored := &v1alpha3.Machine{}
+	if err := utilconversion.UnmarshalData(src, restored); err == nil {
```

Comment:
This seems to ignore all errors, should we not ignore errors that are related to the `json.Unmarshal` call?

Reply:
Added check